_includes/assets/js/indicatorModelHelpers.js

```diff
@@ -0,0 +1,539 @@
+      }, this)[this.UNIT_COLUMN];
+    },
+    getDataByUnit: function(data, unit) {
+      if (!unit) {
+        return data;
+      }
+      return data.filter(function(row) {
+        return row[this.UNIT_COLUMN] === unit;
+      }, this);
+    },
+    getDataBySelectedFields: function(data, selectedFields) {
+      return data.filter(function(row) {
+        return selectedFields.some(function(field) {
+          return field.values.includes(row[field.field]);
+        });
+      });
+    },
+    getDataMatchingCombination: function(data, combination, selectableFields) {
+      return data.filter(function(row) {
+        return selectableFields.every(function(field) {
+          return row[field] === combination[field];
+        });
+      });
+    },
+    getUnitFromStartValues: function(startValues) {
+      var match = startValues.find(function(startValue) {
+        return startValue.field === this.UNIT_COLUMN;
+      }, this);
+      return (match) ? match.value : false;
+    },
+    selectFieldsFromStartValues: function(startValues, selectableFields) {
+      if (!startValues) {
+        return [];
+      }
+      var specialFields = this.nonFieldColumns();
+      var allowedStartValues = startValues.filter(function(startValue) {
+        var normalField = !specialFields.includes(startValue.field);
+        var allowedField = selectableFields.includes(startValue.field)
+        return normalField && allowedField;
+      });
+      var valuesByField = {};
+      allowedStartValues.forEach(function(startValue) {
+        if (!(startValue.field in valuesByField)) {
+          valuesByField[startValue.field] = [];
+        }
+        valuesByField[startValue.field].push(startValue.value);
+      });
+      return Object.keys(valuesByField).map(function(field) {
+        return {
+          field: field,
+          values: valuesByField[field],
+        };
+      });
+    },
+    selectMinimumStartingFields: function(data, fieldNames) {
+      // We filter our full dataset to only those fields.
+      var filteredData = data.filter(function(row) {
+        return fieldNames.some(function(fieldName) {
+          return row[fieldName];
+        });
+      });
+      // We then sort the data by each field. We go in reverse order so that the
+      // first field will be highest "priority" in the sort.
+      fieldNames.reverse().forEach(function(fieldName) {
+        filteredData = filteredData.sort(function(a, b) {
+          return a[fieldName] - b[fieldName];
+        });
+      });
+      // But actually we want the top-priority sort to be the "size" of the
+      // rows. In other words we want the row with the fewest number of fields.
+      filteredData = filteredData.sort(function(a, b) {
+        return Object.keys(a).length - Object.keys(b).length;
+      });
+      // Convert to an array of objects with 'field' and 'values' keys, omitting
+      // any special columns.
+      var specialColumns = this.nonFieldColumns();
```

Comment:
Stick with one naming for these types of columns. "Special columns" or "non field columns"